include/libpmemfile-posix.h

```diff
@@ -158,9 +157,47 @@ extern "C" {
 typedef struct pmemfilepool PMEMfilepool;
 typedef struct pmemfile_file PMEMfile;
 
+typedef unsigned pmemfile_mode_t;
+typedef unsigned pmemfile_uid_t;
+typedef unsigned pmemfile_gid_t;
+typedef long long pmemfile_ssize_t;
+typedef long long pmemfile_off_t;
+typedef unsigned long pmemfile_nlink_t;
+typedef long pmemfile_blksize_t;
+typedef long pmemfile_blkcnt_t;
+typedef unsigned long pmemfile_dev_t;
+typedef unsigned long pmemfile_ino_t;
+
+struct pmemfile_timespec
+{
+	long tv_sec;
+	long tv_nsec;
+};
+
+struct pmemfile_stat
+{
+	pmemfile_dev_t st_dev;
+	pmemfile_ino_t st_ino;
+	pmemfile_nlink_t st_nlink;
+	pmemfile_mode_t st_mode;
+	pmemfile_uid_t st_uid;
+	pmemfile_gid_t st_gid;
+	int __pad0;
+	pmemfile_dev_t st_rdev;
+	pmemfile_off_t st_size;
+	pmemfile_blksize_t st_blksize;
+	pmemfile_blkcnt_t st_blocks;
```

Comment:
The thing is blkcnt_t and blksize_t are signed. We are just providing portable types.